contracts/SwapAndBridge.sol

```diff
@@ -0,0 +1,150 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/USSSpokePoolInterface.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./Lockable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
+
+/// @title Interface for making arbitrary calls during swap
+interface IAggregationExecutor {
+    /// @notice propagates information about original msg.sender and executes arbitrary data
+    function execute(address msgSender) external payable; // 0x4b64e492
+}
+
+// Grabbed from source code of Optimism V5 router: https://optimistic.etherscan.io/address/0x1111111254eeb25477b68fb85ed929f73a960582#code
+// - readable source: https://vscode.blockscan.com/optimism/0x1111111254eeb25477b68fb85ed929f73a960582
+interface I1InchAggregationRouterV5 {
+    struct SwapDescription {
+        IERC20 srcToken;
+        IERC20 dstToken;
+        address payable srcReceiver;
+        address payable dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+    }
+
+    /**
+    /// @notice Performs a swap, delegating all calls encoded in `data` to `executor`.
+    /// @dev router keeps 1 wei of every token on the contract balance for gas optimisations reasons. 
+    /// This affects first swap of every token by leaving 1 wei on the contract.
+    /// @param executor Aggregation executor that executes calls described in `data`
+    /// @param desc Swap description
+    /// @param permit Should contain valid permit that can be used in `IERC20Permit.permit` calls.
+    /// @param data Encoded calls that `caller` should execute in between of swaps
+    /// @return returnAmount Resulting token amount
+    /// @return spentAmount Source token amount     
+    */
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+/**
+ * @title SwapAndBridge
+ * @notice Allows caller to swap tokens on a chain and bridge them via Across atomically.
+ */
+contract SwapAndBridge is Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    USSSpokePoolInterface public immutable spokePool;
+    I1InchAggregationRouterV5 public immutable oneInchRouter;
```

Comment:
Is this interface the same on every chain?

Reply:
Not sure yet, need to test more outside of Optimism